### PR TITLE
Use netip in netdever API

### PR DESCRIPTION
**Pull request by soypat**
netip is a full rework of the `net.IP` type. It solves a wide variety of issues with the net.IP type. It is well documented [here](https://tailscale.com/blog/netaddr-new-ip-type-for-go/).

### CODE REVIEW

**Review by scottfeldman — APPROVED** [review]
LGTM.  Will need corresponding changes in tinygo-driver's netdever interface.

File: netdev.go [hunk 1]
```diff
@@ -48,18 +49,18 @@ type netdever interface {
 
 	// GetHostByName returns the IP address of either a hostname or IPv4
 	// address in standard dot notation
-	GetHostByName(name string) (IP, error)
+	GetHostByName(name string) (netip.Addr, error)
 
 	// GetIPAddr returns IP address assigned to the interface, either by
```
Comment (scottfeldman) [inline netdev.go]: // Addr returns IP address...

**Comment by scottfeldman** [comment]
Added matching PR tinygo-org/drivers#622 to change existing drivers to match netdever interface changes to netip.

File: netdev.go (hunk 1 above)
Comment (soypat) [reply netdev.go]: Fixed!

**Comment by deadprogram** [comment]
Thanks for working on this @soypat and for review @scottfeldman now merging.

**Pull request merged**